### DISCUSSION TITLE
How to provide a config file?

### DISCUSSION THREAD

**Discussion by talksickcs** [v1993/evdevhook2#19] [Q&A] [answered]
Version: AppImage 1.0.1 x86_64
OS: SteamOS 3.5.5 (Arch based)

I know nothing is more frustrating for FOSS developers than people asking boneheaded questions that waste time, but I have been trying on my own for hours to no avail.

I am trying to use this on the SteamDeck to expose a DualSense's motion data. I am also running SteamDeckGyroDSU to do the same thing for the SteamDeck's motion, but this also uses Port 26760. Where do I place a config file and what do I name it, such that the AppImage reads it and changes the port?

I apologize if this is so simple that it would be silly to include in the documentation, I am a noob!

**Comment by v1993** [comment, marked as answer]
Path to the config file is passed as the first argument to the program, so if you launch from terminal it's like

```bash
./evdevhook2-v1.0.1-x86_64.AppImage /path/to/config
```

Do you think it would be a good idea to default to, say, `~/.config/evdevhook2/default.ini` if no config file is specified?

**Reply by talksickcs** [reply]
Thanks for the reply, I will definitely make use of it tomorrow!

>Do you think it would be a good idea to default to, say, ~/.config/evdevhook2/default.ini if no config file is specified?

Yes, I think setting a default config file is good as well as generating this default config if none is found, containing a bit of instructive text. Another thing to consider is creating a new release as a .service file, similar to how SteamDeckGyroDSU works with environment variables. This is probably overkill though, since the framework for providing a config is already in place and systemd is capable of launching an AppImage on startup the same way as a .service.
